### PR TITLE
Remove Unused IndexShardOperationPermits#blockOperations

### PR DESCRIPTION
**Pull request by original-brownbear**
Follow up to #64098, this method was only used in tests now so this PR removes it.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/shard/IndexShardOperationPermits.java [hunk 1]
```diff
@@ -88,30 +87,6 @@ public void close() {
...
-        delayOperations();
-        try (Releasable ignored = acquireAll(timeout, timeUnit)) {
-            onBlocked.run();
-        } finally {
-            releaseDelayedOperations();
-        }
-    }
-
     /**
      * Immediately delays operations and on another thread waits for in-flight operations to finish and then acquires all permits. When all
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/index/shard/IndexShardOperationPermits.java:91]:
```suggestion
     * Immediately delays operations and uses the {@link executor} to wait for in-flight operations to finish and then acquires all permits. When all
```

**Review by DaveCTurner — COMMENTED** [review]
Suggest also renaming `asyncBlockOperations -> blockOperations` since the `async` is redundant (and misleading if using `SAME`).

**Comment by original-brownbear** [comment]
Thanks David, fixed name and Javadoc :)

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Thanks David!

**Pull request merged**